### PR TITLE
Updated docstrings for multiples modules.

### PR DESCRIPTION
**Pull request by reaal-dev**
Update docstring that had a TODO for multiple modules in a major part in the description of classes and headers attributes

### CODE REVIEW

File: dpkt/icmp.py [hunk 1]
```diff
@@ -76,11 +76,17 @@
 class ICMP(dpkt.Packet):
     """Internet Control Message Protocol.
 
-    TODO: Longer class information....
+    The Internet Control Message Protocol (ICMP) is a supporting protocol in the Internet protocol suite.
+    It is used by network devices, including routers, to send error messages and operational information
+    indicating success or failure when communicating with another IP address, for example, when an error
+    is indicated when a requested service is not available or that a host or router could not be reached.
```
Comment (obormot) [inline dpkt/icmp.py:82]: "when an error?" looks like a typo/extra word

File: dpkt/netflow.py [hunk 2]
```diff
@@ -13,11 +13,15 @@
 class NetflowBase(dpkt.Packet):
     """Base class for Cisco Netflow packets.
 
-    TODO: Longer class information....
+    NetFlow is a feature that was introduced on Cisco routers around 1996 that provides the ability to collect IP
+    network traffic as it enters or exits an interface. By analyzing the data provided by NetFlow, a network
+    administrator can determine things such as the source and destination of traffic, class of service, and the causes
+    of congestion. A typical flow monitoring setup (using NetFlow) consists of three main components.
```
Comment (obormot) [inline dpkt/netflow.py:19]: > A typical flow monitoring setup (using NetFlow) consists of three main components.

irrelevant here, IMO

File: dpkt/radiotap.py [hunk 3]
```diff
@@ -62,11 +62,16 @@
 class Radiotap(dpkt.Packet):
     """Radiotap.
 
-    TODO: Longer class information....
+    In this section we'll have a look how the IEEE 802.11 protocol is implemented in libtins. libtins has great support
```
Comment (obormot) [inline dpkt/radiotap.py:65]: looks like a copy-pasta..   please rework this description

File: dpkt/pppoe.py [hunk 4]
```diff
@@ -21,11 +21,19 @@
 class PPPoE(dpkt.Packet):
     """PPP-over-Ethernet.
 
-    TODO: Longer class information....
+    The Point-to-Point Protocol over Ethernet (PPPoE) is a network protocol for encapsulating Point-to-Point Protocol
+    (PPP) frames inside Ethernet frames. It appeared in 1999, in the context of the boom of DSL as the solution for
+    tunneling packets over the DSL connection to the ISP's IP network, and from there to the rest of the Internet.
+    A 2005 networking book noted that "Most DSL providers use PPPoE, which provides authentication, encryption, and
```
Comment (obormot) [inline dpkt/pppoe.py:27]: > A 2005 networking book noted that "Most DSL providers use PPPoE, which provides authentication, encryption, and compression." 

Irrelevant here, IMO

File: dpkt/udp.py [hunk 5]
```diff
@@ -12,11 +12,17 @@
...
-    TODO: Longer class information....
+    User Datagram Protocol (UDP) is one of the core members of the Internet protocol suite.
+    With UDP, computer applications can send messages, in this case referred to as datagrams,
+    to other hosts on an Internet Protocol (IP) network. Prior communications are not required
+    in order to set up communication channels or data paths.
 
     Attributes:
         __hdr__: Header fields of UDP.
-        TODO.
+            sport: (int): (2 bytes)
```
Comment (obormot) [inline dpkt/udp.py:22]: Missing field descriptions here

File: dpkt/ip6.py [hunk 6]
```diff
@@ -20,11 +20,20 @@
...
+    Internet Protocol version 6 (IPv6) is the most recent version of the Internet Protocol (IP),
+    the communications protocol that provides an identification and location system for computers
+    on networks and routes traffic across the Internet. IPv6 was developed by the Internet Engineering
+    Task Force (IETF) to deal with the long-anticipated problem of IPv4 address exhaustion,
+    and is intended to replace IPv4.
 
     Attributes:
         __hdr__: Header fields of IPv6.
-        TODO.
+            _v_fc_flow: (int): Version (4 bits), Traffic Class (6+2 bits) and Flow Label (20 bits).
```
Comment (obormot) [inline dpkt/ip6.py:31]: `_v_fc_flow` is an internal ("private") name for the attribute. 
The fields should be documented separately, e.g.
```
v (int): Version (4 bits)
fc (int): Traffic Class (6+2 bits)
flow (int): Flow Label (20 bits)
```

File: dpkt/ip.py [hunk 7]
```diff
@@ -18,11 +18,22 @@ def get_ip_proto_name(p):
...
 
-    TODO: Longer class information....
+    The Internet Protocol (IP) is the network layer communications protocol in the Internet protocol suite
+    for relaying datagrams across network boundaries. Its routing function enables internetworking, and
+    essentially establishes the Internet.
 
     Attributes:
         __hdr__: Header fields of IP.
-        TODO.
+        _v_hl: (int): Version (4 bit) For IPv4, this is always equal to 4 and Internet Header Length (IHL) (4 bit)
```
Comment (obormot) [inline dpkt/ip.py:27]: `_v_hl` is an internal ("private") name for the attribute. 
The fields should be documented separately, e.g.
```
v (int): Version (4 bits)
hl (int): Internet Header Length (IHL) (4 bits)
```

File: dpkt/ip.py [hunk 8]
```diff
@@ -18,11 +18,22 @@ def get_ip_proto_name(p):
...
+        tos: (int): Type of service. (1 byte)
+        len: (int): Total Length. Defines the entire packet size in bytes, including header and data.(2 bytes)
+        id: (int): Identification. Uniquely identifying the group of fragments of a single IP datagram. (2 bytes)
+        _flags_offset: (int): Flags (3 bit) and Fragment offset (13 bit)
```
Comment (obormot) [inline dpkt/ip.py:31]: see comment for `_v_hl`, same applies.  also 3 bit -> 3 bits.   13 bit -> 13 bits

**Review by obormot — CHANGES_REQUESTED** [review]
Minor fixes requested here and there, great job overall!

**Comment by obormot** [comment]
@reaal-dev Just checking in. Are you planning to address the review comments? We won't be able to merge before that.  Thanks!

File: dpkt/ah.py [hunk 9]
```diff
@@ -11,7 +11,8 @@
 class AH(dpkt.Packet):
     """Authentication Header.
 
-    TODO: Longer class information....
+    The Authentication Header (AH) protocol provides data origin authentication, data integrity, and replay protection.
+    However, AH does not provide data confidentiality, which means that all of your data is sent in the clear.
```
[the remark this answers is not in the document]
Comment (reaal-dev) [reply dpkt/ah.py:15]: Removed 2nd sentence

File: dpkt/dpkt.py [hunk 10]
```diff
@@ -122,23 +125,23 @@ class Packet(_MetaPacket("Temp", (object,), {})):
...
-    >>> foo.bar
-    3
-    >>> foo.baz
-    'quux'
-    >>> foo.foo = 7
-    >>> foo.baz = 'whee'
-    >>> foo
-    Foo(baz='whee', foo=7, bar=3)
-    >>> Foo('hello, world!')
+    class Foo(Packet):
```
[the remark this answers is not in the document]
Comment (reaal-dev) [reply dpkt/dpkt.py:128]: Removed changes.

File: dpkt/cdp.py [hunk 11]
```diff
@@ -27,12 +27,19 @@
...
+    Cisco equipment, such as the operating system version and IP address.
+
     See more on
     https://en.wikipedia.org/wiki/Cisco_Discovery_Protocol
 
     Attributes:
         __hdr__: Header fields of CDP.
-        #TODO
+            version: (int): CDP protocol version. (1 byte)
+            ttl: (int): TIme to live. The amount of time in seconds that a receiver should retain the information
```
[the remark this answers is not in the document]
Comment (reaal-dev) [reply dpkt/cdp.py:40]: Fixed

File: dpkt/dhcp.py [hunk 12]
```diff
@@ -103,11 +103,33 @@
...
+                either Request(1) or Reply(2). (1 byte)
+            hrd: (int): Hardware type. This field specifies the type of hardware address and its value is 1 for Ethernet
+                and is assigned based on the value used in ARP header. (1 byte)
+            hln: (int): Hardware Length. This tells the number of bytes used to hold the MAC address
+                and is usually 6. (1 byte)
+            hops: (int): Hops. This field stores the number of Hops /relays this message has passed.
+                The sender of message set this is 0 (1 byte)
+            xid: (int): Transaction ID. Random value set by the client so the DHCP server can use this same value when
+                 sending a response to the client to identify the messages between them from other messages. (4 bytes)
+            secs: (int): Seconds. This is second elapsed since client started the address acquisition or
```
[the remark this answers is not in the document]
Comment (reaal-dev) [reply dpkt/dhcp.py:122]: Corrected

File: dpkt/dns.py [hunk 13]
```diff
@@ -163,6 +172,8 @@ def opcode(self, v):
 
     @property
     def aa(self):
+        """int: Authoritative Answer. 1 bit.
```
[the remark this answers is not in the document]
Comment (reaal-dev) [reply dpkt/dns.py:175]: Removed "int:"

File: dpkt/dns.py [hunk 14]
```diff
@@ -174,6 +185,7 @@ def aa(self, v):
 
     @property
     def tc(self):
+        """int: Truncated. 1 bit. Indicates that only the first 512 bytes of the reply was returned."""
```
[the remark this answers is not in the document]
Comment (reaal-dev) [reply dpkt/dns.py:188]: Removed "int:"

File: dpkt/http.py [hunk 15]
```diff
@@ -83,11 +83,16 @@ def parse_body(f, headers):
 class Message(dpkt.Packet):
     """Hypertext Transfer Protocol headers + body.
 
-    TODO: Longer class information....
+    HTTP messages are how data is exchanged between a server and a client. There are two types of messages: requests
+    sent by the client to trigger an action on the server, and responses, the answer from the server. HTTP messages are
+     composed of textual information encoded in ASCII, and span over multiple lines. In HTTP/1.1, and earlier versions
+     of the protocol, these messages were openly sent across the connection. In HTTP/2, the once human-readable message
```
[the remark this answers is not in the document]
Comment (reaal-dev) [reply dpkt/http.py:89]: Removed lines

File: dpkt/icmp.py (hunk 1 above)
Comment (reaal-dev) [reply dpkt/icmp.py:82]: Removed example

File: dpkt/netflow.py (hunk 2 above)
Comment (reaal-dev) [reply dpkt/netflow.py:19]: Removed "A typical..."

File: dpkt/radiotap.py (hunk 3 above)
Comment (reaal-dev) [reply dpkt/radiotap.py:65]: Removed description for now

File: dpkt/pppoe.py (hunk 4 above)
Comment (reaal-dev) [reply dpkt/pppoe.py:27]: Removed

File: dpkt/udp.py (hunk 5 above)
Comment (reaal-dev) [reply dpkt/udp.py:22]: Added descriptions

File: dpkt/dhcp.py [hunk 16]
```diff
@@ -103,11 +103,33 @@
...
 
-    TODO: Longer class information....
+    The Dynamic Host Configuration Protocol (DHCP) is a network management protocol used on Internet Protocol (IP)
+    networks for automatically assigning IP addresses and other communication parameters to devices connected
+    to the network using a client–server architecture.
 
     Attributes:
         __hdr__: Header fields of DHCP.
-        TODO.
+            op: (int): Operation. This is called operation field and it identify the message type
```
[the remark this answers is not in the document]
Comment (reaal-dev) [reply dpkt/dhcp.py:112]: Replaced by RFC 2131 definition

File: dpkt/ip6.py (hunk 6 above)
Comment (reaal-dev) [reply dpkt/ip6.py:31]: Added as follows. Please advise.

_v_fc_flow:
     v (int): Version (4 bits)
     fc (int): Traffic Class (6+2 bits)
     flow (int): Flow Label (20 bits)

File: dpkt/ip.py (hunk 7 above)
Comment (reaal-dev) [reply dpkt/ip.py:27]: Added in format described for _v_fc_flow in ip6

File: dpkt/ip.py (hunk 8 above)
Comment (reaal-dev) [reply dpkt/ip.py:31]: Added in format described for _v_fc_flow in ip6

**Comment by reaal-dev** [comment]
Thanks for the reminder

File: dpkt/ip6.py (hunk 6 above)
Comment (obormot) [reply dpkt/ip6.py]: Looks good, except for the order. _v_fc_flow fields should appear before plen, not at the end. Same for the ip header

**Comment by obormot** [comment]
Most everything looks good now. Please address the order of fields in ip and ip6 and we're good to merge.

**Review by obormot — APPROVED** [review]

**Pull request merged**